Hyperspace

lalexgap opened the pull request.

lalexgap committed:
add hyperspace config

lalexgap committed:
fix up ts config files

lalexgap committed:
cors workaround for localhost

lalexgap committed:
upgrade ethers and client-schema

lalexgap committed:
switch to type 2 transactions

lalexgap committed:
handle hyperspace chain id

lalexgap committed:
new version of ethers doesnt like 0x0

lalexgap committed:
check chainId instead of network version

lalexgap committed:
remove preinstall

geoknee commented on .env.local-hyperspace:
```diff
@@ -0,0 +1,69 @@
+## simple-hub, xstate-wallet
+CONSENSUS_APP_ADDRESS = '0xeB1170bf49fac482fA296C98Ac04024e8a2d6519'
+
+### NitroAdjudicator and ETHAssetHolder deployed by George Knee, by uploading files to https://remix.ethereum.org/ and using compiler 0.6.10+commit.00c0fcaf, optimization ON. Source code verified on https://goerli.etherscan.io/ .
```
Let's remove stale comments (copypasta) and references to Goerli.

geoknee commented on packages/iframe-channel-provider/src/postmessage-service.ts:
```diff
@@ -40,6 +40,10 @@ export class PostMessageService {
     this.attempts += 1;
 
     logger.info({message}, 'Sending message (attempt %s)', this.attempts);
+    // TODO: This lets us get around CORS restrictions when using different ports on localhost
+    if (corsUrl.startsWith('http://localhost')) {
+      corsUrl = '*';
+    }
```
Can you explain this a little more?

geoknee commented on packages/xstate-wallet/scripts/start.js:
```diff
@@ -53,7 +53,11 @@ void (async () => {
 
   process.env.TARGET_NETWORK = getNetworkName(process.env.CHAIN_NETWORK_ID);
-
+  // TODO: Devtools doesn't support hyperspace yet
+  if (Number(process.env.CHAIN_NETWORK_ID) === 3141) {
+    process.env.TARGET_NETWORK = 'hyperspace';
+    console.log("Using 'hyperspace' as the target network");
+  }
```
How much work to fixup devtools? We make this change in many places...

geoknee commented on packages/xstate-wallet/src/chain.ts:
```diff
@@ -347,7 +347,7 @@ export class ChainWatcher implements Chain {
     const response = await this.signer.sendTransaction({
       ...convertNitroTransactionRequest(transactionRequest),
-      gasPrice: GAS_PRICE
+      maxPriorityFeePerGas: GAS_PRICE
```
Nit: Is this a good way to migrate the transaction type? Here's a good reference https://www.liquity.org/blog/how-eip-1559-changes-the-transaction-fees-of-ethereum . 

`GAS_PRICE` is defined on L244 with some reference to Goerli network. 

In the short term, we can do whatever seems to work. But we should have an eye on things like this which may break once deployed to Filecoin mainnet.

I think one alternative is to query the chain first to get a `maxPriorityFeePerGas`.

lalexgap committed:
hyperspace versions

lalexgap committed:
add prod env vars

lalexgap committed:
larger cache size to avoid warning

lalexgap committed:
use heroku url for tracker

lalexgap committed:
correct explorer link for hyperspace

geoknee committed:
ignore mesages with data = ""

- I am not sure why we get these messages
- the resulting error is fixed in an upstream version of the package (thru dependence on client-api-schema, where bug was fixed)
- this seems like a very harmless workaround for now

geoknee committed:
Merge pull request #5 from statechannels/ignore-empty-string

ignore mesages with data = ""

geoknee committed:
bump version number

geoknee committed:
remove typo

lalexgap committed:
support for mockingInfuraProvider for hyperspace
This should allow us to use the persistent seeder with hyperspace

geoknee committed:
Update UI to show TFIL not ETH

geoknee committed:
Revert "Update UI to show TFIL not ETH"

This reverts commit 7a1090a149e9ff69af996229d843f8d69d088afa.

geoknee committed:
CHANGE ETH TO FIL

geoknee committed:
Apply suggestions from code review

geoknee committed:
Apply suggestions from code review

geoknee committed:
Merge pull request #7 from statechannels/fil-ui

Fil UI

lalexgap committed:
update to new firebase
